### PR TITLE
add timers for various phases of completion requests

### PR DESCRIPTION
**Pull request by froydnj**
### Motivation

We can see in our metrics that completion requests are relatively slow, but we have no insight into where that time is being spent.  This PR attempts to change that by adding timers for parts of responding to a completion request.

### Test plan

See included automated tests.

### CODE REVIEW

File: main/lsp/requests/completion.cc [hunk 1]
```diff
@@ -849,89 +850,105 @@ vector<unique_ptr<CompletionItem>> CompletionTask::getCompletionItems(LSPTypeche
 
     // ----- methods -----
```
Comment (aprocter) [inline main/lsp/requests/completion.cc:851]: Do we want a timer for the `// ----- keywords -----` stuff above this?

**Review by aprocter — APPROVED** [review]
Nice!

File: main/lsp/requests/completion.cc (hunk 1 above)
Comment (froydnj) [reply main/lsp/requests/completion.cc:851]: Good question.  I thought about adding it, and then decided that this was pretty static: 30 or so prefix matches on a table of keywords *should* be pretty quick and not worth measuring?

**Comment by froydnj** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_LHkTCJuGtNDB55
→ https://go/builds/bui_LHkTiBQP5UgOSr

File: main/lsp/requests/completion.cc (hunk 1 above)
Comment (aprocter) [reply main/lsp/requests/completion.cc:851]: Works for me!

**Pull request merged**